wagtail/search/backends/base.py

```diff
@@ -298,7 +300,7 @@ def add_bulk(self, model, obj_list):
     def delete(self, obj):
         raise NotImplementedError
 
-    def search(self, query, model_or_queryset, fields=None, operator=None, order_by_relevance=True):
+    def search(self, query, model_or_queryset, fields=None, operator=None, order_by_relevance=True, partial_match=True, query_compiler_class=None):
```

Comment:
Passing a `query_compiler_class` in that method seems very weird to me. Do we expect users to use it on the backend? This also gives some inconsistency with the `…QuerySet.search()` method that doesn’t provide this argument.

Maybe the searching logic could be moved into a separate method that both `search` and `autocomplete` would call while passing their own `query_compiler_class`.

Reply:
Just refactored based on your comment, thanks!